examples/text-generation/run_lm_eval_quat_and_msr.py

```diff
@@ -0,0 +1,199 @@
+                    "use_flash_attention": self.options.use_flash_attention,
+                    "flash_attention_recompute": self.options.flash_attention_recompute,
+                    "flash_attention_causal_mask": self.options.flash_attention_causal_mask,
+                }
+            )
+        if args.warmup:
+            self.warm_up()
+
+    def warm_up(self):
+        for bucket_size in reversed(self.buckets):
+            inps = torch.ones((self._batch_size, bucket_size), dtype=torch.int64)
+            self._model_call(inps)
+            pass
+
+    @property
+    def eot_token_id(self):
+        return self.model.config.eos_token_id
+
+    @property
+    def max_length(self):
+        return self.buckets[-1]
+
+    @property
+    def max_gen_toks(self):
+        raise NotImplementedError()
+
+    @property
+    def batch_size(self):
+        return self._batch_size
+
+    @property
+    def device(self):
+        # We need to do padding ourselves, otherwise we'll end up with recompilations
+        # Returning 'cpu' to keep tensors on CPU in lm_eval code
+        return "cpu"
+
+    def tok_encode(self, string):
+        return self.tokenizer.encode(string)
+
+    def tok_decode(self, tokens):
+        return self.tokenizer.decode(tokens)
+
+    def _model_generate(self, context, max_length, eos_token_id):
+        raise NotImplementedError()
+
+    def find_bucket(self, length):
+        return [b for b in self.buckets if b >= length][0]
+
+    def _model_call(self, inps):
+        bs, seq_length = inps.shape
+        padding_length = 0
+        if self.options.static_shapes:
+            bucket_length = self.find_bucket(seq_length)
+            if self.options.use_cache and self.options.reuse_cache:
+                self.model.allocate_kv_cache(bs, bucket_length + 1, bucket_length)
+            padding_length = bucket_length - seq_length
+            inps = F.pad(inps, (0, padding_length), value=self.model.config.pad_token_id)
+        logits = self.model(inps.to(self._device), **self.model_inputs)["logits"].cpu()
+
+        if self.options.static_shapes and padding_length > 0:
+            logits = logits[:, :-padding_length, :]
+        logits = logits.to(torch.float32)
+        return logits
+
+
+def main():
+    args = setup_lm_eval_parser()
+    model, tokenizer, generation_config = initialize_model(args, logger)
+
+    lm_tasks = lm_eval.tasks.get_task_dict(args.tasks)
+    with torch.no_grad():
+        lm = HabanaModelAdapter(tokenizer, model, args, generation_config)
+
+    eval_start_measure = time.perf_counter()
+    results = lm_eval.evaluator.evaluate(lm, lm_tasks, limit=args.limit_iters)
+    if args.device == "hpu":
+        import habana_frameworks.torch.hpu as torch_hpu
+        torch_hpu.synchronize()
+    eval_end_measure = time.perf_counter()
+
+    results["args"] = vars(args)
+    results["duration_msr"] = eval_end_measure - eval_start_measure
```

Comment:
suggest to add print of the measure results also

Reply:
added